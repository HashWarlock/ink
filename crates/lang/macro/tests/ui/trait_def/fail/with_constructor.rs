use ink_lang as ink;

#[ink::trait_definition]
pub trait WithConstructor {
    #[ink(constructor)]
    fn new() -> Self;
}

fn main() {}
