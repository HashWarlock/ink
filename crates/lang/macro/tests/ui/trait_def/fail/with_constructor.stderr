error: ink! trait definitions must not have constructors
 --> $DIR/with_constructor.rs:5:5
  |
5 | /     #[ink(constructor)]
6 | |     fn new() -> Self;
  | |_____________________^
